src/renderers/shared/stack/reconciler/__tests__/ReactCompositeComponentState-test.js

```diff
@@ -213,7 +242,9 @@ describe('ReactCompositeComponent-state', () => {
       // unmountComponent()
       // state is available within `componentWillUnmount()`
       ['componentWillUnmount', 'blue'],
-    ].join('\n'));
+    );
+
+    expect(stateListener.mock.calls.join('\n')).toEqual(expected.join('\n'));
   });
```

Comment:
It would be nice to have a unit test that covers the case you fixed with the new side-effect type. I.e. `setState({}, cb);` where `shouldComponentUpdate` bails out so there is no `componentDidUpdate`.